encoding/json/decode.go

```diff
@@ -1266,6 +1279,34 @@ func (d *Decoder) decodeType(valueJSON any, results typeDecodingResults) cadence
 			d.gauge,
 			d.decodeType(obj.Get(typeKey), results),
 		)
+	case "Restriction":
+		// Backwards-compatibility for format <v1.0.0:
+		if d.backwardsCompatible {
+			restrictionsValue := obj.Get(restrictionsKey)
+			typeValue := obj.Get(typeKey)
+			return d.decodeRestrictedType(
+				typeValue,
+				toSlice(restrictionsValue),
+				results,
+			)
+		} else {
```

Comment:
this just runs the `default` logic and i just copied it over. @turbolent what would you like me to return instead? This func doesnt really let me return an err

Reply:
alternative option is to do it at the default section, ex.

- remove `case "restriction"

- add this to `default`:

```
default: 
    if kindValue == "Restriction" {
        // Backwards-compatibility for format <v1.0.0:
		if d.backwardsCompatible {
			restrictionsValue := obj.Get(restrictionsKey)
			typeValue := obj.Get(typeKey)
			return d.decodeDeprecatedRestrictedType(
				typeValue,
				toSlice(restrictionsValue),
				results,
			)
		}
    }
    // the default case
    simpleType, ok := simpleTypes[kindValue]
		if ok {
			return simpleType
		}
		fieldsValue := obj.Get(fieldsKey)
		typeIDValue := toString(obj.Get(typeIDKey))
		initValue := obj.Get(initializersKey)
		return d.decodeNominalType(
			obj,
			kindValue,
			typeIDValue,
			toSlice(fieldsValue),
			toSlice(initValue),
			results,
		)
   
}
```

Reply:
I'd keep the case. In it, check if backward-compatibility is enabled. If not, just panic saying it's an invalid/unsupported kind. (At some point it would be nice to refactor this file and return error values)